File: AccuPay/Domain/Paystubs/Adjustment.vb
Option Strict On

Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Global.AccuPay.Entity

    <Table("paystubadjustment")>
    Public Class Adjustment

        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property RowID As Integer?

        Public Property OrganizationID As Integer?

        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property Created As Date

        Public Property CreatedBy As Integer?

        <DatabaseGenerated(DatabaseGeneratedOption.Computed)>
        Public Property LastUpd As Date?

        Public Property LastUpdBy As Integer?

        Public Property ProductID As Integer?

        Public Property PaystubID As Integer?

        <Column("PayAmount")>
        Public Property Amount As Decimal

        Public Property Comment As String

        Public Property IsActual As Boolean

        <ForeignKey("PaystubID")>
        Public Overridable Property Paystub As Paystub

        <ForeignKey("ProductID")>
        Public Overridable Property Product As Product

    End Class

End Namespace

File: Accupay.Data/Entities/Adjustment.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace AccuPay.Data.Entities
{
    [Table("paystubadjustment")]
    public class Adjustment : OrganizationalEntity, IAdjustment
    {
        public int? ProductID { get; set; }

        public int? PaystubID { get; set; }

        [Column("PayAmount")]
        public decimal Amount { get; set; }

        public string Comment { get; set; }

        public bool IsActual { get; set; }

        [ForeignKey("PaystubID")]
        public virtual Paystub Paystub { get; set; }

        [ForeignKey("ProductID")]
        public virtual Product Product { get; set; }

        public bool Is13thMonthPay { get; set; }

        public IAdjustment Clone()
        {
            return new Adjustment()
            {
                RowID = RowID,
                OrganizationID = OrganizationID,
                CreatedBy = CreatedBy,
                LastUpdBy = LastUpdBy,
                ProductID = ProductID,
                PaystubID = PaystubID,
                Amount = Amount,
                Comment = Comment,
                IsActual = IsActual,
                Paystub = Paystub,
                Product = Product,
                Is13thMonthPay = Is13thMonthPay,
            };
        }
    }
}
